packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py

```diff
@@ -351,11 +360,11 @@ async def upload_file(
             )
         except (r_clone.RCloneFailedError, exceptions.S3TransferError) as exc:
             log.error("The upload failed with an unexpected error:", exc_info=True)
-            if upload_links:
-                # abort the upload correctly, so it can revert back to last version
-                await _abort_upload(session, upload_links, reraise_exceptions=False)
-                log.warning("Upload aborted")
+            await _abort_upload(session, upload_links, reraise_exceptions=False)
             raise exceptions.S3TransferError from exc
+        except CancelledError:
+            await _abort_upload(session, upload_links, reraise_exceptions=False)

```

Comment:
ok I see now. Can you instead of doing a function that returns right away just check for the upload_links before calling it? what is the point of calling a fct for aborting with empty data? this is wasted fct calls

Reply:
it makes more sense to check the upload_links here, since this is here that they are created. I find that not clear to do it in the separate fct.

Reply:
OK, I was not 100% convinced by this, and you have a point. Will check here in both branches of the exception handling and then raise.